airflow/models/dag.py

```diff
@@ -462,8 +462,10 @@ def __init__(
         # set timezone from start_date
         tz = None
         if start_date and start_date.tzinfo:
-            tzinfo = None if start_date.tzinfo else settings.TIMEZONE
-            tz = pendulum.instance(start_date, tz=tzinfo).timezone
+            if start_date.tzinfo is None:
```

Comment:
this is checked in previous line 464 and confirmed it is valid, does 465 still make sense?